static/app/views/alerts/rules/issue/details/ruleDetails.tsx

```diff
@@ -61,9 +63,47 @@ class AlertRuleDetails extends AsyncComponent<Props, State> {
     });
   }
 
-  componentDidUpdate(prevProps: Props) {
+  async componentDidUpdate(prevProps: Props) {
     const {organization: prevOrg, params: prevParams} = prevProps;
-    const {organization: currOrg, params: currParams} = this.props;
+    const {organization: currOrg, params: currParams, location} = this.props;
+
+    const {ruleId, projectId: projectSlug} = currParams;
+
+    const rule = this.state.rule;
+    const urlParams = new URLSearchParams(location.search);
+    if (urlParams.get('mute') === '1' && rule && !rule.snooze) {
+      try {
+        await this.api.requestPromise(
+          `/projects/${currOrg.slug}/${projectSlug}/rules/${ruleId}/snooze/`,
+          {
+            method: 'POST',
+            data: {
+              target: 'me',
+            },
+          }
+        );
+
+        urlParams.delete('mute');
+        browserHistory.push({
+          pathname: location.pathname,
+          query: urlParams,
+        });
+
+        this.onSnooze({
```

Comment:
You might want to add a `addSuccessMessage` to show a toast when this happens. Otherwise the user might not know that something actually happened. You could either do it just for this case or for every time the alert gets muted/unmuted